components/net/http_cache.rs

```diff
@@ -563,6 +580,132 @@ impl HttpCache {
         }
     }
 
+    /// Get the cache entry corresponding to this request
+    pub fn get_entry(&mut self, request: &Request) -> Option<HttpCacheEntry> {
```

Comment:
How about we just keep it a `RwLock` for now, since everything else on `HttpState` uses one?

https://github.com/servo/servo/blob/9138d1dfed2dc2c28ab2af0052e009202a2ebc09/components/net/http_loader.rs#L64

Reply:
sure, bit of a waste, but not a huge problem.